lib/platform/linux-i2c.c

```diff
@@ -374,15 +375,26 @@ static void i2c_gas_write(struct switchtec_dev *dev, void __gas *dest,
 	uint8_t tag;
 	uint8_t status;
 	uint8_t retry_count = 0;
+	uint8_t status_retry_count = 0;
 
 	do {
+		status_retry_count = 0;
 		tag = get_tag(idev);
 		i2c_gas_data_write(dev, dest, src, n, tag);
-		status = i2c_gas_write_status_get(dev, tag);
-		if (status == 0 || status == GAS_TWI_MRPC_ERR)
+		do {
+			usleep(5000);
```

Comment:
A 5ms delay for every write seems like it would slow it down a ton.... Is this really necessary? We only sleep for 1ms below if it fails.... I2C is supposed to have clock stretching if the device we are communicating with isn't fast enough... does that not work for this hardware?